### PR TITLE
[SPARK-37283][SQL][FOLLOWUP] Avoid trying to store a table which contains timestamp_ntz types in Hive compatible format

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This is PR is to avoid trying to store a table which contains `timestamp_ntz` types in Hive compatible format.
In the current master, Spark tries to store such a table in Hive compatible format first, but it doesn't support `timestamp_ntz`.
As a result warning is logged like as follows though it's finally stored in Spark specific format.
```
CREATE TABLE tbl1(a TIMESTAMP_NTZ) USING Parquet
...
21/11/22 21:57:18 WARN HiveExternalCatalog: Could not persist `default`.`tbl1` in a Hive compatible way. Persisting it into Hive metastore in Spark SQL specific format.
org.apache.hadoop.hive.ql.metadata.HiveException: java.lang.IllegalArgumentException: Error: type expected at the position 0 of 'timestamp_ntz' but 'timestamp_ntz' is found.
        at org.apache.hadoop.hive.ql.metadata.Hive.createTable(Hive.java:869)
        at org.apache.hadoop.hive.ql.metadata.Hive.createTable(Hive.java:874)
        at org.apache.spark.sql.hive.client.Shim_v0_12.createTable(HiveShim.scala:614)
        at org.apache.spark.sql.hive.client.HiveClientImpl.$anonfun$createTable$1(HiveClientImpl.scala:554)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at org.apache.spark.sql.hive.client.HiveClientImpl.$anonfun$withHiveState$1(HiveClientImpl.scala:304)
        at org.apache.spark.sql.hive.client.HiveClientImpl.liftedTree1$1(HiveClientImpl.scala:235)
        at org.apache.spark.sql.hive.client.HiveClientImpl.retryLocked(HiveClientImpl.scala:234)
        at org.apache.spark.sql.hive.client.HiveClientImpl.withHiveState(HiveClientImpl.scala:284)
        at org.apache.spark.sql.hive.client.HiveClientImpl.createTable(HiveClientImpl.scala:552)
        at org.apache.spark.sql.hive.HiveExternalCatalog.saveTableIntoHive(HiveExternalCatalog.scala:506)
        at org.apache.spark.sql.hive.HiveExternalCatalog.createDataSourceTable(HiveExternalCatalog.scala:404)
        at org.apache.spark.sql.hive.HiveExternalCatalog.$anonfun$createTable$1(HiveExternalCatalog.scala:273)
...
```
### Why are the changes needed?

To fix the confusing behavior.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Modified the test added in #34551

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @beliefer

**Comment by sarutak** [comment]
cc: @cloud-fan too.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala [hunk 1]
```diff
@@ -1411,6 +1411,7 @@ object HiveExternalCatalog {
 
   private[spark] def isHiveCompatibleDataType(dt: DataType): Boolean = dt match {
     case _: AnsiIntervalType => false
+    case _: TimestampNTZType => false
```
Comment (cloud-fan) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala:1414]: We need to make a decision here. Storing both ltz and ntz as hive timestamp is also an option. In fact, the hive timestamp is ntz, so storing ntz as hive timestamp is actually correct.

Comment (sarutak) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala:1414]: > We need to make a decision here. Storing both ltz and ntz as hive timestamp is also an option. In fact, the hive timestamp is ntz, so storing ntz as hive timestamp is actually correct.

Ah, O.K, I misunderstood. So if we decide to store both ltz and ntz to Hive table, this change is not necessary right?

**Comment by beliefer** [comment]
> cc @beliefer

Thank you for ping me.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala (hunk 1 above)
Comment (beliefer) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala:1414]: Do we decide to not support nth in Hive?

Comment (sarutak) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala:1414]: Not yet. We are discussing.
https://github.com/apache/spark/pull/34647#discussion_r754237594

**Comment by cloud-fan** [comment]
I think the decision is: table schema with NTZ is not hive-compatible. It's the safest decision as it won't have any regressions.

This PR LGTM

**Comment by sarutak** [comment]
@cloud-fan Thank you for re-visiting this PR!
I'll run GA again just in case.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**